internal/handlers/common/aggregations/operators/operators.go

```diff
@@ -16,24 +16,53 @@
 package operators
 
 import (
-	"context"
+	"fmt"
 
 	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 )
 
-// newAccumulatorFunc is a type for a function that creates an accumulation operator.
-type newAccumulatorFunc func(expression *types.Document) (Accumulator, error)
+// newOperatorFunc is a type for a function that creates a standard aggregation operator.
+type newOperatorFunc func(expression *types.Document) (Operator, error)
 
-// Accumulator is a common interface for aggregation accumulation operators.
-type Accumulator interface {
-	// Accumulate documents and returns the result of applying operator.
-	Accumulate(ctx context.Context, groupID any, in []*types.Document) (any, error)
+// Operator is a common interface for standard aggregation operators.
+type Operator interface {
+	// Process document and returns the result of applying operator.
+	Process(in *types.Document) (any, error)
 }
 
-// GroupAccumulators maps supported accumulators of $group aggregation stage.
-var GroupAccumulators = map[string]newAccumulatorFunc{
+// Get returns operator for provided value.
+func Get(value any, stage, key string) (Operator, error) {
+	operatorDoc, ok := value.(*types.Document)
+	if !ok || operatorDoc.Len() == 0 {
+		return nil, lazyerrors.New(
+			fmt.Sprintf("The field '%s' must be an object", key),
+		)
+	}
```

Comment:
This is not covered in test and I'm quite sure that operators don't need to be document. For example `$type` takes `any`. It's possible I'm misunderstand what value is though :see_no_evil: